### PR TITLE
docs: warn about rpc api being private

### PR DESCRIPTION
**Pull request by lidel**
@BigLep this PR adds the warning we discussed today for https://docs.ipfs.io/reference/http/api/:

> 

> 

Feel free to adjust the language in `tools/http-api-docs/markdown.go` and `docs/reference/http/api.md` 
(the former is used for generating the latter every time a new go-ipfs release ships, so we need to update both places)

### CODE REVIEW

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

File: docs/how-to/command-line-quick-start.md [hunk 1]
```diff
@@ -98,6 +98,12 @@ ipfs daemon
 
 Make a note of the TCP ports you receive. If they are different, use yours in the commands below.
 
+::: danger NEVER EXPOSE THE RPC API TO THE PUBLIC INTERNET
+
+The API port provides admin-level access to your IPFS node, see [RPC API docs](/reference/http/api/) for more information.
```
Comment (BigLep) [inline docs/how-to/command-line-quick-start.md:103]:
```suggestion
The API port provides admin-level access to your IPFS node.  See [RPC API docs](/reference/http/api/) for more information.
```

File: docs/how-to/run-ipfs-inside-docker.md [hunk 2]
```diff
@@ -17,12 +17,18 @@ You can run IPFS inside Docker to simplify your deployment processes, as well as
...
-1. Start a container running ipfs and expose ports `4001`, `5001` and `8080`:
+1. Start a container running ipfs and expose ports `4001` (P2P TCP/QUIC transports), `5001` (RPC API) and `8080` (Gateway):
 
     ```shell
     docker run -d --name ipfs_host -v $ipfs_staging:/export -v $ipfs_data:/data/ipfs -p 4001:4001 -p 4001:4001/udp -p 127.0.0.1:8080:8080 -p 127.0.0.1:5001:5001 ipfs/go-ipfs:latest
     ```
 
+    ::: danger NEVER EXPOSE THE RPC API TO THE PUBLIC INTERNET
+
+    The API port provides admin-level access to your IPFS node, see [RPC API docs](/reference/http/api/) for more information.
```
Comment (BigLep) [inline docs/how-to/run-ipfs-inside-docker.md:28]:
```suggestion
    The API port provides admin-level access to your IPFS node.  See [RPC API docs](/reference/http/api/) for more information.
```

File: docs/reference/http/api.md [hunk 3]
```diff
@@ -6,16 +6,64 @@ description: HTTP RPC API reference for IPFS daemon.
...
 
 In many cases, using this RPC API is preferable to embedding IPFS directly in your program — it allows you to maintain peer connections that are longer lived than your app and you can keep a single IPFS node running instead of several if your app can be launched multiple times. In fact, the `ipfs` CLI commands use this RPC API when operating in online mode.
 
-::: tip
-This document was autogenerated from go-ipfs. For issues and support, check out the [http-api-docs](https://github.com/ipfs/ipfs-docs/tree/main/tools/http-api-docs) repository on GitHub.
+::: danger NEVER EXPOSE THE RPC API TO THE PUBLIC INTERNET
+
+The RPC API provides admin-level access to your IPFS node, including `/api/v0/config`.
+
+It is bound to `localhost` by default on purpose, you should never expose it to the public internet, just like you would never expose an SQL database or other backend service.
```
Comment (BigLep) [inline docs/reference/http/api.md:64]:
```suggestion
It is bound to `localhost` by default on purpose.  You should never expose it to the public internet, just like you would never expose a SQL database or other backend service.
```

File: tools/http-api-docs/markdown.go [hunk 4]
```diff
@@ -24,16 +24,64 @@ description: HTTP RPC API reference for IPFS daemon.
...
 
 In many cases, using this RPC API is preferable to embedding IPFS directly in your program — it allows you to maintain peer connections that are longer lived than your app and you can keep a single IPFS node running instead of several if your app can be launched multiple times. In fact, the `+"`ipfs`"+` CLI commands use this RPC API when operating in online mode.
 
-::: tip
-This document was autogenerated from go-ipfs. For issues and support, check out the [http-api-docs](https://github.com/ipfs/ipfs-docs/tree/main/tools/http-api-docs) repository on GitHub.
+::: danger NEVER EXPOSE THE RPC API TO THE PUBLIC INTERNET
+
+The RPC API provides admin-level access to your IPFS node, including `+"`/api/v0/config`"+`.
+
+It is bound to `+"`localhost`"+` by default on purpose, you should never expose it to the public internet, just like you would never expose an SQL database or other backend service.
```
Comment (BigLep) [inline tools/http-api-docs/markdown.go:82]:
```suggestion
It is bound to `+"`localhost`"+` by default on purpose. You should never expose it to the public internet, just like you would never expose a SQL database or other backend service.
```

**Review by BigLep — APPROVED** [review]
Awesome - thanks @lidel .  Minor suggestions.

**Review by guseggert — APPROVED** [review]

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

**Pull request merged**